### ISSUE TITLE
override.cfg gets pulled into project.godot on reload when external modifications are detected

### ISSUE THREAD

**Issue by dozingpip** [godotengine/godot#92583]
### Tested versions

3.5.3.stable.mono

### System information

Windows 10 - Godot 3.5.3.stable.mono

### Issue description

Related to https://github.com/godotengine/godot/issues/30912 , but not quite the same.
When Godot reloads the project due to external modification, any override.cfg changes are applied directly to project.godot. External modification can happen frequently if say, I keep the project open when I do a git pull for changes from my co-developer. ( my co-dev is https://github.com/godotengine/godot/issues/30912#issuecomment-2140663081)
His theory as to what the problem is: "There's a parameter in the project load function "p_ignore_override" which is supposed to be set to true when loading from the editor. But they forgot to set that parameter to true specifically when reloading the file due to external modifications."

### Steps to reproduce

1. Open Node2D.tscn in editor
2. Open Node2D.tscn in a text editor
3. Modify the tscn somehow, ie change the of "Node2D" to "Node2Dtest" or something, and save
4. Go back to the editor, pick "Reload" when asked
5. project.godot file has been modified to pull in the modified window test size I have in override.cfg.

### Minimal reproduction project (MRP)

[minRepro.zip](https://github.com/user-attachments/files/15509837/minRepro.zip)

**Comment by KoBeWi** [comment]
I just had this happen in 4.3 beta1, but I have no idea how. It's related to reloading, but I couldn't reproduce it with the MRP and it doesn't seem to happen consistently.

> There's a parameter in the project load function "p_ignore_override" which is supposed to be set to true when loading from the editor. But they forgot to set that parameter to true specifically when reloading the file due to external modifications.

That's not the case both on master and in 3.x

https://github.com/godotengine/godot/blob/705b7a0b0bd535c95e4e8fb439f3d84b3fb4f427/editor/editor_node.cpp#L1178

Unless the reload is done elsewhere.

**Comment by ace24713** [comment]
> That's not the case both on master and in 3.x

Yes, it is the case on both master and in 3.x. Unless I'm mistaken `p_ignore_override` is the implicit 4th parameter to ProjectSettings::setup, which defaults to `false` on that line of code you linked.  The 3rd one that's given as `true` there is `p_upwards`

see the declaration here:
https://github.com/godotengine/godot/blob/705b7a0b0bd535c95e4e8fb439f3d84b3fb4f427/core/config/project_settings.h#L186

**Comment by KoBeWi** [comment]
I know why I couldn't reproduce. The reload dialog buttons do the opposite of what they say.
